builtin.py

```diff
@@ -66,24 +74,28 @@ class Location(Node):
 
 # Actions include the player for reference (augment player attributes)
 class Action(Node):
-    def __init__(self, title: str, desc: str, player: Player):
+    def __init__(self, title: str, desc: str, coins: str, player: Player):
         self.player = player
-        super().__init__(title, desc)
+        super().__init__(title, desc, coins)
 
     def on_select(self):
         print(self.player.summary())
+        self.player.coins_earned += self.coins
+        super(Action, self).on_select()
 
 
 class Fight(Action):
     def on_select(self):
         self.player.hp -= 25
         self.player.xp += 10
+        self.player.coins_earned += 50
 
         super().on_select()
 
 
 class Run(Action):
     def on_select(self):
+        self.player.coins_earned += 10
         self.player.xp = max(self.player.xp - 10, 0)
 
-        super().on_select()
\ No newline at end of file
+        super().on_select()
```

Comment:
This is not an issue with your code, but moreso a comment on mine. We can probably get rid of these classes, since the Action class does pretty much everything we want to at this point